File: BackEnd/backend/src/main/java/com/fitapp/backend/application/dto/sport/SportRequest.java
package com.fitapp.backend.application.dto.sport;

import jakarta.validation.constraints.NotBlank;
import lombok.Data;
import java.util.Map;

@Data
public class SportRequest {
    
    @NotBlank(message = "El nombre del deporte es obligatorio")
    private String name;

    private Map<String, String> parameterTemplate;
    
    private String category;
}

File: FrontEnd/app/src/main/java/com/fitapp/appfit/response/sport/request/SportRequest.kt
package com.fitapp.appfit.response.sport.request

import com.google.gson.annotations.SerializedName

data class SportRequest(
    @SerializedName("name") val name: String,
    @SerializedName("parameterTemplate") val parameterTemplate: Map<String, String>? = null,
    @SerializedName("sourceType") val sourceType: String? = "USER_CREATED"
)